draft-ietf-tls-esni.md

```diff
@@ -520,9 +524,11 @@ to be padded using TLS record layer padding.
 ## Handling the server response {#handle-server-response}
 
 As described in {{server-behavior}}, the server MAY either accept ECH and use
-ClientHelloInner or reject it and use ClientHelloOuter. However, there is no
-indication in ServerHello of which one the server has done and the client must
-therefore use trial decryption in order to determine this.
+ClientHelloInner or reject it and use ClientHelloOuter. In handling the server's
+response, the client's first step is to determine which was used. To do so, it
+inspects the last 8 bytes of the ServerHello.random: if this string is equal to
```

Comment:
Nit: I would refactor the sentence beginning with "To do so". Something like this?

> The client presumes acceptance if the last 8 bytes of the ServerHello.random are equal to the ClientHelloInner's "ech" extension payload. Otherwise, it presumes rejection.

Reply:
Good idea! Done.